packages/ui/src/memberships/hooks/usePostParents.ts

```diff
@@ -0,0 +1,20 @@
+import { ApiRx } from '@polkadot/api'
+import { useMemo } from 'react'
+
+import { useApi } from '@/common/hooks/useApi'
+import { ForumPostParentsFragment, useGetForumPostParentsQuery } from '@/forum/queries'
+
+export const usePostParents = (id: string) => {
+  const { data, loading } = useGetForumPostParentsQuery({ variables: { where: { id } } })
+  const { api } = useApi()
+  const { threadId, categoryId } = useMemo(() => asPostParents(data?.forumPostByUniqueInput, api), [data, loading])
+  return { threadId, categoryId, parentsLoading: loading }
+}
+
+const asPostParents = (fields: ForumPostParentsFragment | null | undefined, api: ApiRx | undefined) =>
+  fields && api
+    ? {
+        threadId: api.createType('ThreadId', fields.thread.id),
+        categoryId: api.createType('CategoryId', fields.thread.category.id),
```

Comment:
This can be done using `ApiRx` or the helper from `@joystream/types`, though the helper can only accept numeric values (rather than a string):
```suggestion
        threadId: createType('ThreadId', Number.parseInt(fields.thread.id)),
        categoryId: createType('CategoryId', Number.parseInt(fields.thread.category.id)),
```
I guess it's a matter of preference

Reply:
Yeah, that doesn't matter TBH. The thing I'm worried here, that this hook might be too specialised and/or it mixes Query-Node types with internal and API types. Usually, we will do them in two steps:

1. In hooks fetching data we translate external types from `...Fields` to our internal types.
2. When touching API we would t[ranslate JS types to Substrate API types](https://github.com/Joystream/pioneer/blob/9a7f4a09b9c4fb41f5105b04eb57219380ac228d/packages/ui/src/memberships/modals/utils.ts#L11)

(we however break that for BN as BN is used to store token values).